### PR TITLE
Statinf/instrumentation

### PR DESCRIPTION
**Pull request by brouxel**
Adding the instrumentation tool first version into the main tool branch

### CODE REVIEW

**Comment by brouxel** [comment]
I validate my own push

**Comment by SlimBenAmor** [comment]
I don't have a deep knowledge in C++ and LLVM to make a relevant review. I tried to test it on my laptop but it did not succeed to compile LLVM even0 with 16Go of RAM directly after a restart. 
My only remark is that we may use another name than "Liebherr" to name variables, functions,... (eg. instead of `PragmaLiebherrStmt` we may use `PragmaTiStmt` or `PragmaTiC2000Stmt` or `PragmaTmsStmt` to mention the used [µC](https://www.ti.com/product/TMS320F28379D)/[architecture](https://www.ti.com/microcontrollers-mcus-processors/c2000-real-time-control-mcus/overview.html)/[compiler](https://www.ti.com/tool/C2000-CGT)).

**Review by SlimBenAmor — COMMENTED** [review]
I don't have a deep knowledge in C++ and LLVM to make a relevant review. I tried to test it on my laptop but it did not succeed to compile LLVM even0 with 16Go of RAM directly after a restart. 
My only remark is that we may use another name than "Liebherr" to name variables, functions,... (eg. instead of `PragmaLiebherrStmt` we may use `PragmaTiStmt` or `PragmaTiC2000Stmt` or `PragmaTmsStmt` to mention the used [µC](https://www.ti.com/product/TMS320F28379D)/[architecture](https://www.ti.com/microcontrollers-mcus-processors/c2000-real-time-control-mcus/overview.html)/[compiler](https://www.ti.com/tool/C2000-CGT)).

**Comment by jordr** [comment]
@brouxel Thanks for all this work. This is some sizeable and complex LLVM code, I don't think I will find the time to give even a shallow review of this.
I'm sorry we don't have the resources to provide peer review with this part of the project at the moment in general, at least until May. I hope this will improve on the long term.

File: clang/include/clang/Basic/TokenKinds.def [hunk 1]
```diff
@@ -943,6 +942,15 @@ ANNOTATION(module_end)
...
 ANNOTATION(header_unit)
 
+// Texas Instrument
+KEYWORD(interrupt                   , KEYALL)
+KEYWORD(cregister                   , KEYALL)
+KEYWORD(reentrant                   , KEYALL)
+KEYWORD(trap                        , KEYALL)
+ALIAS("__interrupt" , interrupt     , KEYALL)
+ALIAS("__cregister", cregister      , KEYALL)
+ALIAS("__restrict" , restrict       , KEYALL)
```
Comment (jordr) [inline clang/include/clang/Basic/TokenKinds.def:952]: @brouxel what about `__volatile` and `__asm`?

Comment (brouxel) [reply clang/include/clang/Basic/TokenKinds.def:952]: They were both be already present: line 671 for the volatile, line 652 for the asm.

**Pull request merged**